Cannot use UriInfo context in resteasy-reactive `@ServerResponseFilter`

### Describe the bug

According to the [RESTeasy Reactive guide](https://quarkus.io/guides/resteasy-reactive#request-or-response-filters), filter parameters can take any [context object](https://quarkus.io/guides/resteasy-reactive#context-objects). However, declaring `UriInfo` as a parameter in a response filter results in a runtime error. (This happens for a few other context objects.)

It seems that `UriInfo` is a valid parameter for `getRequestFilterResultHandles` but not for [`getResponseFilterResultHandles`](https://github.com/quarkusio/quarkus/blob/4757a61b87556c456724ef63f9b8fd8068b8eacd/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/filters/CustomFilterGenerator.java#L472-L510).
https://github.com/quarkusio/quarkus/blob/4757a61b87556c456724ef63f9b8fd8068b8eacd/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/filters/CustomFilterGenerator.java#L289-L293

I'm not sure if this is a documentation or implementation issue.

### Expected behavior

It's possible to access `UriInfo` in a response filter with `ContainerRequestContext#getUriInfo()`. Ideally, response filters could directly declare `UriInfo` without error. 

### Actual behavior

```
Caused by: java.lang.RuntimeException: Parameter 'uriInfo' of method 'getFilter of class 'org.acme.Filters' is not allowed
        at io.quarkus.resteasy.reactive.server.deployment.CustomFilterGenerator.getResponseFilterResultHandles(CustomFilterGenerator.java:511)
        at io.quarkus.resteasy.reactive.server.deployment.CustomFilterGenerator.generateStandardContainerResponseFilter(CustomFilterGenerator.java:424)
        at io.quarkus.resteasy.reactive.server.deployment.CustomFilterGenerator.generateContainerResponseFilter(CustomFilterGenerator.java:336)
        at io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveScanningProcessor.handleCustomAnnotatedMethods(ResteasyReactiveScanningProcessor.java:369)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:887)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:829)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

### How to Reproduce?

Create a filter class which declares `javax.ws.rs.core.UriInfo` as a parameter. 

```java
package org.acme;

import org.jboss.resteasy.reactive.server.ServerResponseFilter;

import javax.ws.rs.container.ContainerResponseContext;
import javax.ws.rs.core.UriInfo;

class Filters{
    @ServerResponseFilter
    public void getFilter(ContainerResponseContext responseContext, UriInfo uriInfo) {
        System.out.println("Request path is: " + uriInfo.getRequestUri().toString());
    }
}
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.5.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/CustomContainerResponseFilter.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/CustomContainerResponseFilter.java
@@ -1,10 +1,12 @@
 package io.quarkus.resteasy.reactive.server.test.customproviders;
 
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import javax.ws.rs.container.ContainerRequestContext;
 import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.core.UriInfo;
 
 import org.jboss.resteasy.reactive.server.ServerResponseFilter;
 import org.jboss.resteasy.reactive.server.SimpleResourceInfo;
@@ -14,10 +16,11 @@ public class CustomContainerResponseFilter {
 
     @ServerResponseFilter
     public void whatever(SimpleResourceInfo simplifiedResourceInfo, ContainerResponseContext responseContext,
-            ContainerRequestContext requestContext, Throwable t) {
+            ContainerRequestContext requestContext, UriInfo uriInfo, Throwable t) {
         assertTrue(
                 PreventAbortResteasyReactiveContainerRequestContext.class.isAssignableFrom(requestContext.getClass()));
         assertNull(t);
+        assertNotNull(uriInfo);
         if (simplifiedResourceInfo != null) {
             responseContext.getHeaders().putSingle("java-method", simplifiedResourceInfo.getMethodName());
         }

--- a/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/filters/CustomFilterGenerator.java
+++ b/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/filters/CustomFilterGenerator.java
@@ -488,6 +488,11 @@ final class CustomFilterGenerator {
                 targetMethodParamHandles[i] = filterMethod.getMethodParam(1);
             } else if (unwrappableTypes.contains(paramDotName)) {
                 targetMethodParamHandles[i] = GeneratorUtils.unwrapObject(filterMethod, rrReqCtxHandle, paramDotName);
+            } else if (URI_INFO.equals(paramDotName)) {
+                GeneratorUtils.paramHandleFromReqContextMethod(filterMethod, rrReqCtxHandle, targetMethodParamHandles,
+                        i,
+                        "getUriInfo",
+                        URI_INFO);
             } else if (RESOURCE_INFO.equals(paramDotName)) {
                 targetMethodParamHandles[i] = getResourceInfoHandle(filterMethod, rrReqCtxHandle);
             } else if (ResteasyReactiveServerDotNames.SIMPLIFIED_RESOURCE_INFO.equals(paramDotName)) {

--- a/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/ServerResponseFilter.java
+++ b/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/ServerResponseFilter.java
@@ -8,6 +8,7 @@ import javax.ws.rs.Priorities;
 import javax.ws.rs.container.ContainerRequestContext;
 import javax.ws.rs.container.ContainerResponseContext;
 import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.UriInfo;
 
 /**
  * When used on a method, then an implementation of {@link javax.ws.rs.container.ContainerResponseContext} is generated
@@ -41,6 +42,7 @@ import javax.ws.rs.container.ResourceInfo;
  * <li>{@link ContainerRequestContext}
  * <li>{@link ContainerResponseContext}
  * <li>{@link ResourceInfo}
+ * <li>{@link UriInfo}
  * <li>{@link SimpleResourceInfo}
  * <li>{@link Throwable} - The thrown exception - or {@code null} if no exception was thrown
  * </ul>

